### ISSUE TITLE
BENTHIC_INVERTS and SAMPLES metadata for zoobenthos samples

### ISSUE THREAD

**Issue by ctsolomon** [MFEh2o/db#33]
The sampleIDs that show up in BENTHIC_INVERTS for EL and WL are associated with three metadataIDs in the SAMPLES table: "crazy.Chiro2015.2016", "zoobenthos.Sample.20110714", and "zoobenthos.Sample.20181114". Crazy Chiro definitely had different metadata than our usual sampling, but for the other samples I think they should maybe all have "zoobenthos.Sample.20110714" as the metadataID in SAMPLES, rather than the 20181114 version; I don't think anything about the sampling changed in 2018, either in actuality or in the "zoobenthos.Sample.20181114" metadata file. We should check on this and resolve as appropriate. It may be that the 2018 metadata file describes some differences in lab processing that should get captured in a zoobenthosLengths metadata description that shows up in the BENTHIC_INVERTS table, and/or that it describes some things that got done incorrectly in 2018 and should get fixed, like using the gridded paper to do length measurements.

**Comment by ctsolomon** [comment]
A related issue, I think - which seems to affect many or all of the 2017 benthic invertebrate samples in Long. These samples show up in the SAMPLES table with a sampleID that includes the "BenthicInvert.Sample.20110714" metadataID bit of the sampleID string, but they show up in the BENTHIC_INVERTEBRATES table with a sampleID that includes the "zoobenthos.Sample.20110714" metadataID bit of the sampleID string.

**Comment by kaijagahm** [comment]
> I don't think anything about the sampling changed in 2018, either in actuality or in the "zoobenthos.Sample.20181114" metadata file. We should check on this and resolve as appropriate. It may be that the 2018 metadata file describes some differences in lab processing that should get captured in a zoobenthosLengths metadata description that shows up in the BENTHIC_INVERTS table, and/or that it describes some things that got done incorrectly in 2018 and should get fixed, like using the gridded paper to do length measurements.

Looks like what changed about sampling was the location of a few of the sites. I'm not sure what to do about this.

The 2018 metadata file does indeed describe differences in lab processing. By "some differences in lab processing that should get captured in a zoobenthosLengths metadata description that shows up in the BENTHIC_INVERTS table", do you mean that metadata regarding the lab processing should be associated with BENTHIC_INVERTS and removed from SAMPLES? If so, would we need two different metadataID's for the changes in length measurement protocols?

It looks like they did use gridded paper for length measurements in 2018. Is that indeed incorrect? What needs to be done to fix this?

**Comment by kaijagahm** [comment]
> A related issue, I think - which seems to affect many or all of the 2017 benthic invertebrate samples in Long. These samples show up in the SAMPLES table with a sampleID that includes the "BenthicInvert.Sample.20110714" metadataID bit of the sampleID string, but they show up in the BENTHIC_INVERTEBRATES table with a sampleID that includes the "zoobenthos.Sample.20110714" metadataID bit of the sampleID string.

Confirmed this. Which is correct? Presumably the sampleID should be the same between the two tables, so that the samples can be related to each other. But if it's true that we can have two different metadataID docs, one for sampling protocol and the other for lab/processing protocol, then how do we choose which one to use in the sampleID?

**Comment by kaijagahm** [comment]
See also #32

**Comment by kaijagahm** [comment]
Another weird thing about "BenthicInvert.Sample.20110714" vs. "zoobenthos.Sample.20110714": Alex's old script bugTableFix_20181018.R has some lines that change the metadataID for all samples taken after 2017-05-09 from "BenthicInvert.Sample.20110714" to "zoobenthos.Sample.20110714". See below [and this is lines 76-79 in the attached script]:

```
#All 2017 samples have the wrong metadataID; batch change all
sammy <- which(samples$dateSample > '2017-05-09' & samples$metadataID == "BenthicInvert.Sample.20110714")
#turn into zoobenthos.Sample.20110714
samples$metadataID[sammy] = "zoobenthos.Sample.20110714"
```

However, when I look at the current version of the SAMPLES table (or, rather, the current version of BENTHIC_INVERT_SAMPLES, but same difference, here), there are still many 2017 samples that have "BenthicInvert.Sample.20110714" as their metadataID.

Why would this be? Did Alex's changes in this script not go through? It's confusing, because other changes he made in this same script do seem to have gone through, so it's not like he didn't run the script at all or didn't finalize the database update. There are some previously missing rows that he successfully filled in in this script, as well as some taxa names that he changed using this script. So either the changes to the metadataID's somehow didn't go through, or they were changed back at some point.

**Comment by kaijagahm** [comment]
I was getting confused with all the different files, so I went through all of them, read them, and took stock of how many samples in each table they're associated with. Here's a summary, with my recommendations about what to do.

**(1) BenthicInvert.Sample.20110714.docx**
- Primarily a field protocol. Gives a pretty detailed description of the sampling scheme, laid out in a table. Also talks about the sampling done with D-nets for isotope/lipid analysis.
- Mentions hand-picking and storage in 70% ethanol, and ID, but does not go into detail about these lab methods. 
- Associated with 206 rows in SAMPLES and 0 rows in BENTHIC_INVERTS. That's as it should be--this metadata only pertains to sampling, not lab processing.
*PROPOSAL: make this the canonical field methods protocol (though could change the name if we like the "zoobenthos" convention better than "BenthicInvert")*

**(2) zoobenthos.Sample.20110519.docx**
- Field protocol only
- Short paragraph: sampling via cores + Ekman grabs. Some sample structure, area information.
- Seems like just a less detailed version of zoobenthos.Sample.20110714.docx
- Associated with 0 rows in SAMPLES and 0 rows in BENTHIC_INVERTS. In fact, this metadataID *only* appears in METADATA, nowhere else in database.
*PROPOSAL: Double check to make sure all the information in here is reflected in a different metadata doc so we're not losing any info. Then delete this file and remove it from METADATA.*

**(3) zoobenthos.Sample.20110714.docx**
- Field protocol
- Contains info from (2), but with some additions. Still only two paragraphs; not as comprehensive as (1).
- As mentioned in #32, should probably treat (1) as canonical where the area measurements differ from (3) because it was updated more recently. 
- Associated with 1241 rows in SAMPLES and 0 rows in BENTHIC_INVERTS.
*PROPOSAL: Should probably get rid of this file (3) and consider (1) to be canonical. Make sure we don't lose any info from (3) in doing this. Can keep the name from (3) if we like it better than (1).*

**(4) zoobenthosLengths.20110519.docx**
- Lab protocol for picking, ID, body size, dry mass, etc. Seems to be the standard file for lab processing of benthic inverts.
- Associated with 0 rows in SAMPLES (good, since this is a lab-only protocol) and 39131 rows in BENTHIC_INVERTS.
*PROPOSAL: No change needed. Keep as is.*

**(5) zoobenthosSample.20181114.docx**
- Both field and lab protocol. Not good! Need to have those be separate.
- Repeats the field protocol from (2)/(3). 
- Most of the protocols in (5) are lab-related. Detailed instructions on how to measure inverts, how to calibrate the scope, use ImageJ, etc.
- Methods use grid paper for length measurements. @ctsolomon mentioned that this method might have been incorrect. 
- Associated with 158 rows in SAMPLES and 6030 rows in BENTHIC_INVERTS.
*PROPOSAL: Remove all mention of field protocols, since those are covered in the newly-consolidated (1)/(3) doc that I'll make, and there don't seem to be any changes to the field protocol. Rows in SAMPLES associated with (5) should be changed to (1)/(3). Change the name of (5) to e.g. zoobenthosLengths.20181114.docx to clarify that this is a lab protocol only, analogous to (4). For the 6030 rows in BENTHIC_INVERTS associated with (5), check whether those are all from 2018. If they are, leave as is. If some of them are from earlier years, change those to be associated with (4) instead. If any of them are from later years, check whether the 2018 methods were indeed used for processing in those later years and change to (4) if not. Since it seems that the 2018 methods were incorrect, should make it clear that they are not to be replicated in other years. Maybe add a note to that effect in the doc itself. @ctsolomon can you clarify which parts of these lab methods were incorrect and which ones were fine, so I can make that note?*

@ctsolomon @joneslabND general question: the table is called BENTHIC_INVERTS (and soon we'll have BENTHIC_INVERT_SAMPLES). The docs here seem to use the word "zoobenthos" instead of "benthic inverts", except for (1). Do we prefer the "zoobenthos" name, even though it doesn't match the db table names? I find the zoobenthos/zooplankton distinction minorly confusing, but if "zoobenthos" is the word usually used, then no problem. Should just be consistent.

**Comment by ctsolomon** [comment]
I agree with your proposals in 1-4. (As always, of course, we should make sure that update notes allow us to trace back overwrites of data).

Re #5 - we have talked about doing some spot-checks of 2018 bug measurements, which might influence how we handle this one. Let's discuss tomorrow.

Re names - I agree it would be a little easier if we used either benthic invert or zoobenthos. I don't think I have a strong preference, we should do whatever makes the most sense. Let's take a vote tomorrow.

**Comment by kaijagahm** [comment]
Going to implement the proposals in (1)-(4). Even though I like the "benthicInverts" name better because it fits with the name of the database tables, I'm going to stick with "zoobenthos" because it won't require overhauling the metadataID's in a huge number of sampleID's.

**Comment by kaijagahm** [comment]
Updating the database as is, before waiting for the 2018 decision--we can add that later. I want to get the BENTHIC_INVERT_SAMPLES table all squared away.

**Comment by kaijagahm** [comment]
Amaryllis has sent me bug samples form 2018 so I can compare lengths to the lengths in the database. Going to match up the samples and run a regression. New script is "2018BugLenghCheck2020_gh33.R"

**Comment by kaijagahm** [comment]
After running regressions, decided to change the metadataID's for the 2018 samples to match the non-2018 samples, since we didn't find any bias or big differences in measurements between Amaryllis's re-measurements and the measurements stored in the database. The R-squared for the head width regression was a lot lower than for the body length regression, but we noted that there are no NA values for either headWidth or bodyLength of the 2018 samples in the database, so we don't think this is a big problem.

Have gone through and updated the appropriate metadataID's, sampleID's, and invertID's in both BENTHIC_INVERTS and BENTHIC_INVERT_SAMPLES. Checked whether these sampleID's appear anywhere else in the database, and they don't, so no additional updates were needed. All details are in the script mentioned above. Also added a note to the main bug processing metadata file to discuss the change. Moved the 2018 bug metadata file into General/Secondary Metadata so we can keep a copy but so that it's detached.

This is ready for meta updates in the metadata script and for a database update.

**Comment by kaijagahm** [comment]
Fixed with v 3.5.2, 2020-10-20

**Comment by kaijagahm** [comment]
I'm reopening this because, concerningly, it looks like the change to the 2018 metadataID's didn't go through for BENTHIC_INVERT_SAMPLES. We still have a whole bunch of samples that have the 2018 metadataID in their metadataID column and in their sampleID's.

This should have been fixed in sections 5 and 6 of the script 2018BugLengthCheck2020_gh33_gh81_gh85.R, and looking back at it I don't see a reason why it wasn't. Maybe the BENTHIC_INVERT_SAMPLES file didn't get properly incorporated into the database update somehow?? I'm so confused.

Or else... did Randi do an update right after this one that involved the BIS table? If so, could she have pulled from the older version of the database? --> No, I just checked VERSION_HISTORY, and that did not happen.

Anyway, I apparently need to re-run this script, or at least re-do the parts of it that made those changes.

**Comment by kaijagahm** [comment]
Good news! I did not majorly mess up here, only minorly. The code that I ran did run successfully, and the change was made in most places. But in my old code, I made the assumption that only samples from 2018 would have the 2018 metadataID. That's a reasonable assumption, but it turns out it was wrong. There were a whole bunch of samples from 2017 that also had the 2018 metadataID, for some reason.

In the script BENTHIC_INVERTS_duplicate_invertIDs_gh88.R, I started off by fixing this problem. Fixed metadataID's in the metadataID column in both BENTHIC_INVERTS and BENTHIC_INVERT_SAMPLES, and noted the change in the updateID's. There were no sampleID's in BENTHIC_INVERTS that had the wrong metadataID in the sampleID, so no change was necessary there. There were 158 sampleID's in BENTHIC_INVERT_SAMPLES that had the wrong metadataID in the sampleID, so I corrected those. I made a lookup table in case I needed to use it to correct these sampleID's in other tables, but luckily they didn't turn up in any other database tables, so that change wasn't needed.

This issue will be closed once I update the database to fix issue #88.
